kolibri/core/notifications/utils.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import unicode_literals
+
+import functools
+
+
+def memoize(fun):
+    """A simple memoize decorator for functions supporting (hashable)
+    positional arguments.
+    It also provides a cache_clear() function for clearing the cache:
+
+    >>> @memoize
+    ... def foo()
+    ...     return 1
+        ...
+    >>> foo()
+    1
+    >>> foo.cache_clear()
+    >>>
+    """
+
+    @functools.wraps(fun)
+    def wrapper(*args, **kwargs):
+        key = (args, frozenset(sorted(kwargs.items())))
+        try:
+            return cache[key]
+        except KeyError:
+            ret = cache[key] = fun(*args, **kwargs)
+            return ret
+
+    def cache_clear():
+        """Clear cache."""
+        cache.clear()
+
+    cache = {}
```

Comment:
@rtibbles if we don't plan to move from using LocMemCache to other kind as redis or memcache, yes, it would be equivalent. Please, confirm me if you prefer me to change it to use Django cache.

Reply:
One nice thing about using the Django cache, even without using a different backend, is that we can let the cache expire regularly, and not risk having stale values stored in the memoize?